### PR TITLE
chore: Use pre-commit, update justfile and fix lints

### LINKED ISSUE #318: chore: Update CI and development docs
We've made multiple changes in CQCL/hugr that should be ported here too.

These include (but are not limited to):
- Simplify CONTRIBUTING.md
  - Add explicit mention for `maturin develop` usage
- Simpler justfile, no `just test` / `just pytest` difference.
- Configure pre-commit
- Update the pr-title workflow
- ...

Most of these should be directly copyable from hugr.

### PR DESCRIPTION
**Pull request by aborgna-q**
Changes are a bit messy since enabling pre-commit required fixing some lints.

The main changes are:
- `justfile`: Brought in the simpler recipes from hugr
- `.pre-commit-config.yaml`: Configured all the checks (Except for mypy, since we are missing typing info on the bindings).

Closes #318 (although I'll also update the ci checks workflow at some point)

### CODE REVIEW

**Comment by aborgna-q** [comment]
Oh, I forgot to update the readme and contributing.md. I'll add a commit

File: README.md [hunk 1]
```diff
@@ -14,10 +14,10 @@ quantum software developers to take advantage of its state of the art
 compilation for many different quantum architectures.
 
 Circuits are represented using the HUGR IR defined in the
-[quantinuum-hugr] crate. TKET2 augments Hugr with  
-* The [`Circuit`] trait, providing a high-level interface for working with HUGRs representing quantum circuits  
-* a HUGR extension with quantum operations  
-* A composable pass system for optimising circuits  
+[quantinuum-hugr] crate. TKET2 augments Hugr with
```
Comment (ss2165) [inline README.md:17]:
```suggestion
[hugr] crate. TKET2 augments Hugr with
```

**Comment by aborgna-q** [comment]
Updated DEVELOPMENT.md with the new instructions.

I also included a section for using maturin to test the library locally in an environment (I got a support request on this before).

**Review by ss2165 — APPROVED** [review]
thanks

**Pull request merged**